### PR TITLE
EIP 2981 Integration + locked downcasting

### PR DESCRIPTION
**Pull request by viraj124**

### CODE REVIEW

File: contracts/JBTiered721Delegate.sol [hunk 1]
```diff
@@ -762,4 +762,21 @@ contract JBTiered721Delegate is IJBTiered721Delegate, JB721Delegate, Ownable {
     _tokenId;
     _tier;
   }
+
+  /**
+    @notice Return royalty info
```
Comment (drgorillamd) [inline contracts/JBTiered721Delegate.sol:767]: Maybe a bool "useRoyalty", so owners could still use their reserved rate without being force to take royalties?

Comment (drgorillamd) [reply contracts/JBTiered721Delegate.sol:767]: (maybe in fc metadata or smth, so it can't be changed)

Comment (viraj124) [reply contracts/JBTiered721Delegate.sol:768]: @drgorillamd made it optional with `JBFundingCycleMetadata.metadata`  have another look when you get time

File: contracts/JBTiered721Delegate.sol [hunk 2]
```diff
@@ -26,7 +27,7 @@ import './structs/JBTiered721Flags.sol';
   Votes: A helper for voting balance snapshots.
   Ownable: Includes convenience functionality for checking a message sender's permissions before executing certain transactions.
 */
-contract JBTiered721Delegate is IJBTiered721Delegate, JB721Delegate, Ownable {
+contract JBTiered721Delegate is IJBTiered721Delegate, IERC2981, JB721Delegate, Ownable {
```
Comment (xBA5ED) [inline contracts/JBTiered721Delegate.sol:30]: Nit: Interfaces should be inherited first (on the right), this makes it less likely we run into `linearization of inheritance graph impossible` issues later on

File: contracts/JBTiered721Delegate.sol [hunk 3]
```diff
@@ -762,4 +763,26 @@ contract JBTiered721Delegate is IJBTiered721Delegate, JB721Delegate, Ownable {
...
+      (, JBFundingCycleMetadata memory _metadata) = IJBController(directory.controllerOf(projectId)).currentFundingCycleOf(projectId);
+      // 4 specifies `useRoyalty` is true
+      if (_metadata.metadata == 4) {
+        JB721Tier memory _tier = store.tier(address(this), _tokenId);
+        return (directory.projects().ownerOf(projectId), PRBMath.mulDiv(
+          _salePrice,
+          _tier.reservedRate,
+          JBConstants.MAX_RESERVED_RATE
+        ));
+      } else return (directory.projects().ownerOf(projectId), 0); // when no riaylties have been enabled
```
Comment (xBA5ED) [inline contracts/JBTiered721Delegate.sol:786]: Nit: Typo in comment ;)

File: contracts/JBTiered721Delegate.sol [hunk 4]
```diff
@@ -762,4 +763,26 @@ contract JBTiered721Delegate is IJBTiered721Delegate, JB721Delegate, Ownable {
...
+    @param _salePrice Sales price
+    @return receiver royalty receiver address
+    @return royaltyAmount Royalty Amount
+  */
+  function royaltyInfo(uint256 _tokenId, uint256 _salePrice) external view override returns (address receiver, uint256 royaltyAmount) {
+      _tokenId; //avoid compiler warning
+      // Get the project's current funding cycle metadata.
+      (, JBFundingCycleMetadata memory _metadata) = IJBController(directory.controllerOf(projectId)).currentFundingCycleOf(projectId);
+      // 4 specifies `useRoyalty` is true
+      if (_metadata.metadata == 4) {
```
Comment (xBA5ED) [inline contracts/JBTiered721Delegate.sol:779]: Q: Won't this clash with some of the other (existing) metadata settings?

Comment (viraj124) [reply contracts/JBTiered721Delegate.sol:779]: I checked the v3 repo didn't find any clashes so think we should be good

File: contracts/JBTiered721Delegate.sol [hunk 5]
```diff
@@ -38,6 +38,11 @@ contract JBTiered721Delegate is JB721Delegate, Ownable, IJBTiered721Delegate, IE
   error RESERVED_TOKEN_MINTING_PAUSED();
   error TRANSFERS_PAUSED();
 
+  //*********************************************************************//
+  // ------------------------- public constants ------------------------ //
+  //*********************************************************************//
+  uint256 public constant MAX_ROYALTY_RATE = 200;
```
Comment (viraj124) [inline contracts/JBTiered721Delegate.sol:44]: should we keep `MAX_ROYALTY_RATE` as `10_000` like `max_reserved_rate` for consistency ?

Comment (mejango) [reply contracts/JBTiered721Delegate.sol:44]: no because it has to fit in a uint8, pretty sure.

Comment (mejango) [reply contracts/JBTiered721Delegate.sol:44]: ill play with storage once all open PRs have been merged

**Review by drgorillamd — APPROVED** [review]

**Pull request merged**